diff --git a/boards/c3dpb/board.mk b/boards/c3dpb/board.mk
new file mode 100644
index 0000000..c17f5fb
--- /dev/null
+++ b/boards/c3dpb/board.mk
@@ -0,0 +1,2 @@
+CHIP_FAMILY = samd51
+CHIP_VARIANT = SAMD51J19A
diff --git a/boards/c3dpb/board_config.h b/boards/c3dpb/board_config.h
new file mode 100644
index 0000000..32e5181
--- /dev/null
+++ b/boards/c3dpb/board_config.h
@@ -0,0 +1,26 @@
+#ifndef BOARD_CONFIG_H
+#define BOARD_CONFIG_H
+
+#define CRYSTALLESS 1
+
+#define VENDOR_NAME "C3DPB"
+#define PRODUCT_NAME "C3DPB"
+#define VOLUME_LABEL "C3DPBBOOT"
+#define INDEX_URL "https://github.com/wRieDen/c3dpb"
+#define BOARD_ID "SAMD51J19A-C3DPB"
+
+#define USB_VID 0x239A
+#define USB_PID 0x0022
+
+#define BOOT_USART_MODULE                 SERCOM5
+#define BOOT_USART_MASK                   APBAMASK
+#define BOOT_USART_BUS_CLOCK_INDEX        MCLK_APBDMASK_SERCOM5
+#define BOOT_USART_PAD_SETTINGS           UART_RX_PAD1_TX_PAD0
+#define BOOT_USART_PAD3                   PINMUX_UNUSED
+#define BOOT_USART_PAD2                   PINMUX_UNUSED
+#define BOOT_USART_PAD1                   PINMUX_PB30D_SERCOM5_PAD1
+#define BOOT_USART_PAD0                   PINMUX_PB31D_SERCOM5_PAD0
+#define BOOT_GCLK_ID_CORE                 SERCOM3_GCLK_ID_CORE
+#define BOOT_GCLK_ID_SLOW                 SERCOM3_GCLK_ID_SLOW
+
+#endif
diff --git a/inc/uf2.h b/inc/uf2.h
index 56aba54..9ffeec8 100644
--- a/inc/uf2.h
+++ b/inc/uf2.h
@@ -46,9 +46,9 @@
 // Check various conditions; best leave on
 #define USE_ASSERT 0 // 188 bytes
 // Enable reading flash via FAT files; otherwise drive will appear empty
-#define USE_FAT 1 // 272 bytes
+#define USE_FAT 0 // 272 bytes
 // Enable index.htm file on the drive
-#define USE_INDEX_HTM 1 // 132 bytes
+#define USE_INDEX_HTM 0 // 132 bytes
 // Enable USB CDC (Communication Device Class; i.e., USB serial) monitor for Arduino style flashing
 #define USE_CDC 1 // 1264 bytes (plus terminal, see below)
 // Support the UART (real serial port, not USB)
@@ -56,7 +56,7 @@
 // Support Human Interface Device (HID) - serial, flashing and debug
 #define USE_HID 1 // 788 bytes
 // Expose HID via WebUSB
-#define USE_WEBUSB 1
+#define USE_WEBUSB 0
 // Doesn't yet disable code, just enumeration
 #define USE_MSC 1
 
